contracts/assemblyscript/nep4/__tests__/main.unit.spec.ts

```diff
@@ -1,138 +1,138 @@
 import { VM, Context } from 'near-sdk-as'
 import * as nonSpec from '../main'
-import {
-  grant_access,
-  revoke_access,
-  transfer,
-  check_access,
-  get_token_owner,
-} from '../main'
-
-const alice = 'alice'
-const bob = 'bob'
-const carol = 'carol'
-
-describe('grant_access', () => {
-  it('grants access to the given account_id for all the tokens that account has', () => {
-    const aliceToken = nonSpec.mint_to(alice)
-
-    Context.setPredecessor_account_id(alice)
-    grant_access(bob)
-
-    Context.setPredecessor_account_id(bob)
-    expect(check_access(alice)).toBe(true)
-  })
-})
-
-describe('revoke_access', () => {
-  it('revokes access to the given `accountId` for the given `tokenId`', () => {
-    // Prevent error `InconsistentStateError(IntegerOverflow)` thrown by near-sdk-rs
-    Context.setStorage_usage(100)
-
-    const aliceToken = nonSpec.mint_to(alice)
 
-    Context.setPredecessor_account_id(alice)
-    grant_access(bob)
+function testNep4Interface(contract: string, accounts: { id: string, token: string }[]): void {
```

Comment:
@willemneal how can I accomplish this sort of array-of-objects-with-specific-shape in AssemblyScript?

Reply:
Needs to be a class. Then either, `Array<T>` or `T[] `.